StarVMC/StarAgmlChecker/StarAgmlChecker.cxx

```diff
@@ -151,7 +141,21 @@ void StarAgmlChecker::Fill( TObjectSet *set, Double_t rmin, Double_t rmax, Doubl
 	if ( gGeoManager->IsOutside() ) startnode = 0;
 
 	matprop = 0;
-	if ( startnode ) matprop = startnode->GetVolume()->GetMaterial()->GetRadLen();        	 
+	if ( startnode ) {
+	  TGeoMaterial* material = startnode->GetVolume()->GetMaterial();
+	  matprop = material->GetRadLen();
+	  TString nm = material->GetName();
+	  nm.ToLower();
+	  if ( nm.Contains("air") ) {                                   // use the name of the material to select air...
+	    if ( false == mScoreAir ) {                                 // user option to ignore air in detector volumes
+	      if ( !name.Contains("CAVE") && !name.Contains("HALL") ) { // will still count air in the cave and hall histograms
+		matprop = 0;                                            // property will not be scored
+	      }
+	    }
+	  }
```

Comment:
For better readability please consider replacing the nested 'if' statements with a single one such as:

```c++
bool condition = conditionA && !conditionB && conditionC && !conditionD;
matprop = 0;
if (condition) {
  matprop = material->GetRadLen();
}
```